<a href="https://colab.research.google.com/github/GHAITHASHQAR/-leri-Olas-l-k-/blob/main/Basic_Probability1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Karşılaşma Problemi

### **Problem** : Bir spor salonuna günde 500 kişi spor yapmak için geliyor. Gelen kişilerin %40'ı kilo vermek istiyor. Sabah saatlerinde (10:00-12:00 saatleri arasında) kulübe 25 kişi geliyor. 
### Sabah saatlerinde kulübe gelen kişilerden kilo vermek isteyen bir kişiyle karşılaşma olasılığımız nedir?

**p (Toplam Kişi)** = 500

**s (Sabah saatlerinde gelen kişi sayısı)** = 25

**f (Kilo Vermek İsteyen Kişi Sayısı)** = 500 * 40 / 100 = 200

*Her kişi ile karşılaşma eşit şanslı olay olursa:*

1 değerinden istenmeyen durumların gerçekleşme olasılığını çıkarırsak sonuca ulaşabiliriz.

**P (kilo vermek isteyen kişi sayısı)** = 1 - C (p-f,s) / C (p,s)

In [ ]:
def faktoriyel(sayi):
  sonuc=1
  for i in range(1,sayi+1):
    sonuc=sonuc*i
  return sonuc
  
def kombinasyon(n,r):
  return faktoriyel(n)/(faktoriyel(n-r)*faktoriyel(r))

In [ ]:
p=500
s=25
f=int(p*0.4)


k=p-f

P=1-kombinasyon(k,s)/kombinasyon(p,s)
print(P)

0.9999981289002444


**Sonuç:** Sabah saatlerinde spor kulübüne gelen kişilerden kilo vermek isteyen bir kişiyle karşılaşma olasılığımız **0.9999981289002444** olarak bulunur.